src/pyodide/python.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * This file is a simplified version of the Pyodide loader:
+ * https://github.com/pyodide/pyodide/blob/main/src/js/pyodide.ts
+ *
+ * In particular, it drops the package lock, which disables
+ * `pyodide.loadPackage`. In trade we add memory snapshots here.
+ */
+
+
+/**
+ * _createPyodideModule and pyodideWasmModule together are produced by the
+ * Emscripten linker
+ */
+import { _createPyodideModule } from "pyodide-internal:pyodide-bundle/pyodide.asm";
+import pyodideWasmModule from "pyodide-internal:pyodide-bundle/pyodide.asm.wasm";
+
+
+/**
+ * The Python and Pyodide stdlib zipped together. The zip format is convenient
+ * because Python has a "ziploader" that allows one to import directly from a
+ * zip file.
+ *
+ * The ziploader solves bootstrapping problems around unpacking: Python comes
+ * with a bunch of C libs to unpack various archive formats, but they need stuff
+ * in this zip file to initialize their runtime state.
+ */
+import stdlib from "pyodide-internal:pyodide-bundle/python_stdlib.zip";
+
+/**
+ * Global variable for the memory snapshot. On the first run we stick a copy of
+ * the heap here, on subsequent runs we can skip bootstrapping Python which is
+ * quite slow. Startup with snapshot is 3-5 times faster than without it.
+ *
+ * In the future we could consider creating a memory snapshot ahead of time so
+ * that latency is more predictable.
+ *
+ * Unfortunately this memory snapshot can easily get to ~30mb depending on the
+ * script. The peak memory usage for Python seems to be reached at startup when
+ * compiling Python source code, so if we use pre-compiled Python code the heap
+ * is much smaller. Unfortunately, in recent versions of Python, pyc has large
+ * inline caches which bloats the size of pyc files by a lot. A special-built
+ * compressor should be able to drop the inline cache space and then calculate
+ * how much space to insert from the opcode metadata.
+ *
+ * Possibly we could make many scripts share the same memory snapshot to get
+ * decent performance/memory usage tradeoff.
+ */
+let memory = undefined;
+
+/**
+ * Much simpified version of the `prepareFileSystem` function here:
```

Comment:
Rather than documenting something as a simplified version of external code I'd prefer a documentation of what does _this_ function do.